src/schema_generators/generators.cljc

```diff
@@ -69,19 +69,21 @@
      (generators/one-of
       (for [o (macros/safe-get s :options)]
         (if-let [g (:guard o)]
-          (generators/such-that g (sub-generator o params))
-          (sub-generator o params))))))
+          (generators/such-that g (sub-generator o params) (:max-retries opts))
+          (sub-generator o params))))
+     (:max-retries opts)))
 
   ;; TODO: this does not currently capture proper semantics of maps with
   ;; both specific keys and key schemas that can override them.
   schema.spec.collection.CollectionSpec
-  (composite-generator [s params]
+  (composite-generator [s params opts]
     (generators/such-that
      (complement (.-pre ^schema.spec.collection.CollectionSpec s))
-     (generators/fmap (:konstructor s) (elements-generator (:elements s) params))))
+     (generators/fmap (:konstructor s) (elements-generator (:elements s) params))
+     (:max-retries opts)))
 
   schema.spec.leaf.LeafSpec
-  (composite-generator [s params]
+  (composite-generator [s _]
```

Comment:
I think this is missing the third argument if you're going that way

Reply:
I will follow the `params` path